utils/src/lib.rs

```diff
@@ -84,9 +84,24 @@ pub fn union_unique(namespace: &[u8], msg: &[u8]) -> Vec<u8> {
     result
 }
 
+/// Compute the modulo of bytes interpreted as a big-endian integer.
+///
+/// This function is used to select a random entry from an array
+/// when the bytes are a random seed.
+pub fn modulo(bytes: &[u8], n: u64) -> u64 {
```

Comment:
This is commonly used to select an entry based on the output of an arbitrary length VRF (we want to use all bytes of randomness rather than truncating to be some index).